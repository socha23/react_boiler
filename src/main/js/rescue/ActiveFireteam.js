import React from 'react'
import {connect} from 'react-redux'

import Panel from '../common/components/Panel'

import {tagDescriptionsByTagId} from '../tags/tagHelpers'
import TagAreaName from '../tags/TagAreaName'
import DistanceBetweenTags from '../tags/DistanceBetweenTags'

const ActiveFireteam = ({
        fireteam,
        fireteamTag,
        targetTag,
        tagDescriptionsByTagId = {},
        }) => <div style={{display: "flex", alignItems: "center", minHeight: 84}}>
    <div style={{flex: 1}}>
        <div style={{display: "flex"}}>


            <div style={{flex: 1}}>
                <span style={{fontWeight: "bold", fontSize: 20, marginRight: 10}}>
                    {fireteam.name}
                </span>
                <TagAreaName tag={fireteamTag} style={{fontSize: 16}}/>
            </div>
            <div style={{fontSize: 20}}>
                <DistanceBetweenTags from={fireteamTag} to={targetTag}/>
            </div>
        </div>
        <div style={{marginTop: 10}}>
            <FireteamTarget fireteamTag={fireteamTag} targetTag={targetTag} tagDescriptionsByTagId={tagDescriptionsByTagId}/>
            <TagAreaName tag={targetTag}/>
        </div>
    </div>
</div>;

const FireteamTarget = ({fireteamTag, targetTag, tagDescriptionsByTagId = {}}) =>
                targetTag ?
                        <span style={{fontSize: 20, marginRight: 10}}>
                            {tagDescriptionsByTagId[targetTag.id]}
                        </span>

                        :
                        <span>Nie ustawiono celu</span>
        ;

const mapStateToProps = (state, ownProps) => ({
    tagDescriptionsByTagId: tagDescriptionsByTagId(state.tags.items, state.artifacts.items, state.fireteams.items),
    fireteamTag: state.tags.itemsById[ownProps.fireteam.tagId],
    targetTag: ownProps.fireteam.targetTagId ? state.tags.itemsById[ownProps.fireteam.targetTagId] : null
});

export default connect(mapStateToProps)(ActiveFireteam);