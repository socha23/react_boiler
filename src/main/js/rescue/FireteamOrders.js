import React from 'react'
import {connect} from 'react-redux'

import Panel from '../common/components/Panel'

import {tagDescriptionsByTagId} from '../tags/tagHelpers'
import DistanceBetweenTags from '../tags/DistanceBetweenTags'
import * as fireteamActions from '../fireteams/fireteamActions'

const FireteamOrders = ({
        fireteam = {},
        fireteamTag = {},
        targetTag = {},
        tagDescriptionsByTagId = {},
        onSetTargetTag = () => {
        }
        }) => <Panel>
    <div style={{minHeight: 100}}>
        {

            fireteam && targetTag && fireteam.id != targetTag.id ?
                    <SetTargetButton fireteam={fireteam} fireteamTag={fireteamTag}
                                     targetTag={targetTag}
                                     tagDescriptionsByTagId={tagDescriptionsByTagId}
                                     onSetTargetTag={onSetTargetTag}/>
                    : <span/>}
    </div>
</Panel>;

const SetTargetButton = ({fireteam, fireteamTag, targetTag, tagDescriptionsByTagId, onSetTargetTag}) =>
        <a className="btn btn-block btn-primary btn-lg" onClick={() => onSetTargetTag(fireteam, targetTag)}>
            <div style={{display: "flex", minHeight: 80, alignItems: 'center'}}>
                <div style={{flex: 1, textAlign: 'left'}}>
                    <i className="glyphicon glyphicon-screenshot" style={{marginRight: 10}}/>
                    {fireteam.name}: {tagDescriptionsByTagId[targetTag.id]}
                </div>
                <div>
                    <DistanceBetweenTags from={fireteamTag} to={targetTag}/>
                </div>
            </div>


        </a>;

const ClearTargetButton = ({fireteam, onSetTargetTag}) =>
        <a className="btn btn-block btn-default" onClick={() => onSetTargetTag(fireteam, null)}>
            Wyczyść cel
        </a>;

const mapStateToProps = (state, ownProps) => ({
    tagDescriptionsByTagId: tagDescriptionsByTagId(state.tags.items, state.artifacts.items, state.fireteams.items),
    fireteamTag: ownProps.fireteam && ownProps.fireteam.id ? state.tags.itemsById[ownProps.fireteam.tagId] : null
});

const mapDispatchToProps = (dispatch) => ({
    onSetTargetTag: (fireteam, targetTag) => dispatch(fireteamActions.setFireteamTargetTag(fireteam, targetTag))
});

export default connect(mapStateToProps, mapDispatchToProps)(FireteamOrders);