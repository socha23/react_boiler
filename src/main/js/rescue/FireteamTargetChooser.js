import React from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'

import TaggedObjectsList from '../tags/TaggedObjectsList'

const RescueTargetsChooser = ({
        selected = {},
        tags = [],
        onSelect = () => {}
    }) =>
        <TaggedObjectsList selected={selected} tags={tags} onSelect={onSelect}/>;

export default RescueTargetsChooser