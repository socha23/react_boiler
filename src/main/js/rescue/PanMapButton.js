import React from 'react'

const PanMapButton = ({onClick}) => <a style={{fontSize: 25}} onClick={e => {e.stopPropagation(); onClick(e)}}>
    <i className="glyphicon glyphicon-map-marker" />

</a>;

export default PanMapButton;