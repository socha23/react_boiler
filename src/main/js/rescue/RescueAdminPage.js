import React from 'react'
import FireteamsBrowserContainer from '../fireteams/FireteamsBrowserContainer'
import PanelWithTitle from '../common/components/PanelWithTitle'

export default () => <div className="container-fluid">
        <div className="row">
            <div className="col-sm-4 colWithSmallerGutter">
                <PanelWithTitle title="Roty straÅ¼ackie">
                    <FireteamsBrowserContainer/>
                </PanelWithTitle>
            </div>
        </div>
</div>
