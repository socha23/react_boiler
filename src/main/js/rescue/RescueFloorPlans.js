import React from 'react'
import {connect} from 'react-redux'

import {getTagsInsideByCoordinateSystemId} from '../tags/selectors'
import {getFloorPlans} from '../maps/selectors'
import {getFireteamAndTargetTags} from '../fireteams/selectors'

import TabPanel, {STYLE_LG} from '../common/components/TabPanel'

import ZoomableFloorPlan from '../maps/ZoomableFloorPlan'
import {STYLE_ANIMATED_GREEN} from '../maps/Line'


let RescueFloorPlan = ({floorPlan, tags, markedTag, onSelect, additionalMargin, targetLines, tagDecorations}) => <ZoomableFloorPlan
                        map={floorPlan}
                        tags={tags}
                        tagDecorations={tagDecorations}
                        selectedTag={markedTag}
                        onClickTag={onSelect}
                        additionalMargin={additionalMargin}
                        lines={targetLines}
                        />;


RescueFloorPlan = connect((state, ownProps) => ({
    tags: getTagsInsideByCoordinateSystemId(state)[ownProps.floorPlan.id],
    tagDecorations: tagDecorations(ownProps.selectedTarget, ownProps.selectedFireteam),
    targetLines: getFireteamAndTargetTags(state, ownProps.floorPlan.id).map(t => line(t.fireteamTag, t.targetTag, STYLE_ANIMATED_GREEN))
}))(RescueFloorPlan);


function tagDecorations(selectedTargetTag, selectedFireteam) {
    const result = {};
    if (selectedTargetTag) {
        result[selectedTargetTag.id] = "pulseGlowBlue"
    }
    if (selectedFireteam) {
        result[selectedFireteam.tagId] = "pulseGlowRed"
    }
    return result
}

function line(fromTag, toTag, style) {
    return {
        fromX: fromTag.position.x,
        fromY: fromTag.position.y,
        toX: toTag.position.x,
        toY: toTag.position.y,
        style: style
    }
}

class RescueFloorPlans extends React.Component {
    static defaultProps = {
        selectedTarget: null,
        selectedFireteam: null,
        floorPlans: [],
        markedTag: null,
        onSelect: () => {},
        additionalMargin: 0
    };

    state = {
        lastSelected: {},
        activeTab: 0
    };

    componentWillReceiveProps = (nextProps) => {
        if (nextProps.markedTag && nextProps.markedTag.id != this.state.lastSelected.id) {
            this.setState({
                lastSelected: nextProps.markedTag,
                activeTab: this.props.floorPlans.findIndex(fp => fp.id == nextProps.markedTag.coordinateSystemId)
            });
        } else if (!nextProps.markedTag) {
            this.setState({
                lastSelected: {}
            });

        }
    };

    onTabChange = (i) => {this.setState({activeTab: i})};

    render = () =>
            <TabPanel
                    {...this.props}
                    tabStyle={STYLE_LG}
                    padding={0}
                    activeTab={this.state.activeTab}
                    onTabChange={this.onTabChange}
                    tabs={
               this.props.floorPlans.map(fp => ({
                    label: fp.name,
                    body: <RescueFloorPlan
                        floorPlan={fp}
                        markedTag={this.props.markedTag}
                        selectedTarget={this.props.selectedTarget}
                        selectedFireteam={this.props.selectedFireteam}
                        onSelect={this.props.onSelect}
                        additionalMargin={this.props.additionalMargin}
                        />
               }))
            }/>
}

export default connect((state) => ({
    floorPlans: getFloorPlans(state)
}))(RescueFloorPlans)
