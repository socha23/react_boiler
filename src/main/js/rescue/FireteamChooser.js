import React from 'react'
import {connect} from 'react-redux'

import Panel from '../common/components/Panel'

import ActiveFireteam from './ActiveFireteam.js'


const OptionalScroll = ({fireteams = [], children}) => fireteams.length <= 2 ? children :
    <div className="scrollPanel" style={{
        height: 114,
        overflowX: "auto",
        overflowY: "hidden"
    }}>
        {children}
    </div>;


export const FireteamChooser = ({
                                    fireteams = [],
                                    selected = null,
                                    onSelect = () => {},
                                    tagsById = {}
                                }) =>
    <OptionalScroll fireteams={fireteams}>
        <div style={{display: "flex", flexDirection: "row", margin: "-5px -5px 0 -5px"}}>
            {fireteams.map(f => {
                    let isSelected = selected && selected.id == f.id;
                    return <div style={{flex: 1, margin: 5}} key={f.id}>
                        <Panel
                            style={{
                                minWidth: 250,
                                backgroundColor: isSelected ? "#f2dede" : "#fff",
                                cursor: "pointer"
                            }}
                            onClick={() => onSelect(f)}
                        >
                            <ActiveFireteam fireteam={f}/>
                        </Panel>
                    </div>
                }
            )}
        </div>
    </OptionalScroll>;

const mapStateToProps = (state) => ({
    tagsById: state.tags.itemsById
});

export default connect(mapStateToProps)(FireteamChooser);