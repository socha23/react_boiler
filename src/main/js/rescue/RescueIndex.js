import React from 'react'
import {Switch, Route, Redirect} from 'react-router'
import {NavLink} from 'react-router-dom'
import LiNavLink from '../common/components/LiNavLink'
import PageTemplate from '../templates/PageTemplate'

import RescueInsidePage from './RescueInsidePage'
import RescueOutsidePage from './RescueOutsidePage'
import RescueAdminPage from './RescueAdminPage'

const menu = <ul>
    <LiNavLink to="/rescue/inside/:id?">Akcja w muzeum</LiNavLink>
    <LiNavLink to="/rescue/outside/:id?">Na zewnÄ…trz</LiNavLink>
    <LiNavLink liStyle={{float: "right", position: "relative", left: -150}} to="/rescue/admin">Administracja</LiNavLink>

</ul>;

const content = <Switch>
    <Route exact path="/rescue"><Redirect to="/rescue/inside"/></Route>
    <Route path="/rescue/inside" component={RescueInsidePage}/>
    <Route path="/rescue/outside" component={RescueOutsidePage}/>
    <Route path="/rescue/admin" component={RescueAdminPage}/>
</Switch>;

export default () => <PageTemplate pageNav={menu} content={content}/>
