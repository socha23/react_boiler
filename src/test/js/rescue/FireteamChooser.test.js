import React from 'react'

import {elemWithProvider} from '../testUtils'

import {FireteamChooserComponent} from 'rescue/FireteamChooser'
import FireteamChooser from "../../../main/js/rescue/FireteamChooser";

const FT_1 = {
    id: "FT_1",
    name: "fireteam 1"
};

const FT_2 = {
    id: "FT_2",
    name: "fireteam 2"
};

const FT_3 = {
    id: "FT_3",
    name: "fireteam 3"
};


function stateWithFireteams(fireteams) {
    return {
        fireteams: {
            items: fireteams
        },
        tags: {
            items: []
        }
    }
}


it('renders correctly', () => {
    expect(elemWithProvider(
        <FireteamChooser fireteams={[FT_1, FT_2]}/>, stateWithFireteams([FT_1, FT_2]
        )
    )).toMatchSnapshot()
});

it('doesnt contain scroll for two fireteams', () => {
    expect(elemWithProvider(
        <FireteamChooser fireteams={[FT_1, FT_2]}/>, stateWithFireteams([FT_1, FT_2])
    ).find(".scrollPanel")).toBeEmpty()
});

it('contains scroll for three fireteams', () => {
    expect(elemWithProvider(
        <FireteamChooser fireteams={[FT_1, FT_2, FT_3]}/>, stateWithFireteams([FT_1, FT_2, FT_3])
    ).find(".scrollPanel")).toBePresent()
});
