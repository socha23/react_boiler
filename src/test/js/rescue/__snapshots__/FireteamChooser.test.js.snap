// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(FireteamChooser)
    fireteams={
      Array [
        Object {
          "id": "FT_1",
          "name": "fireteam 1",
        },
        Object {
          "id": "FT_2",
          "name": "fireteam 2",
        },
      ]
    }
  >
    <FireteamChooser
      dispatch={[Function]}
      fireteams={
        Array [
          Object {
            "id": "FT_1",
            "name": "fireteam 1",
          },
          Object {
            "id": "FT_2",
            "name": "fireteam 2",
          },
        ]
      }
    >
      <OptionalScroll
        fireteams={
          Array [
            Object {
              "id": "FT_1",
              "name": "fireteam 1",
            },
            Object {
              "id": "FT_2",
              "name": "fireteam 2",
            },
          ]
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "margin": "-5px -5px 0 -5px",
            }
          }
        >
          <div
            key="FT_1"
            style={
              Object {
                "flex": 1,
                "margin": 5,
              }
            }
          >
            <Panel
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "cursor": "pointer",
                  "minWidth": 250,
                }
              }
            >
              <div
                className="panel panel-default"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "cursor": "pointer",
                    "minWidth": 250,
                  }
                }
              >
                <div
                  className="panel-body"
                  style={
                    Object {
                      "padding": 10,
                    }
                  }
                >
                  <Connect(ActiveFireteam)
                    fireteam={
                      Object {
                        "id": "FT_1",
                        "name": "fireteam 1",
                      }
                    }
                  >
                    <ActiveFireteam
                      dispatch={[Function]}
                      fireteam={
                        Object {
                          "id": "FT_1",
                          "name": "fireteam 1",
                        }
                      }
                      fireteamTag={null}
                      targetTag={null}
                    >
                      <div
                        style={
                          Object {
                            "alignItems": "center",
                            "display": "flex",
                            "minHeight": 84,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <span
                                style={
                                  Object {
                                    "fontSize": 20,
                                    "fontWeight": "bold",
                                    "marginRight": 10,
                                  }
                                }
                              >
                                fireteam 1
                              </span>
                              <Connect(TagAreaNameComponent)
                                style={
                                  Object {
                                    "fontSize": 16,
                                  }
                                }
                                tag={null}
                              >
                                <TagAreaNameComponent
                                  dispatch={[Function]}
                                  style={
                                    Object {
                                      "fontSize": 16,
                                    }
                                  }
                                  tag={null}
                                >
                                  <span />
                                </TagAreaNameComponent>
                              </Connect(TagAreaNameComponent)>
                            </div>
                            <div
                              style={
                                Object {
                                  "fontSize": 20,
                                }
                              }
                            >
                              <DistanceBetweenTags
                                from={null}
                                to={null}
                              >
                                <span />
                              </DistanceBetweenTags>
                            </div>
                          </div>
                          <div
                            style={
                              Object {
                                "marginTop": 10,
                              }
                            }
                          >
                            <FireteamTarget
                              fireteamTag={null}
                              targetTag={null}
                            >
                              <span>
                                Nie ustawiono celu
                              </span>
                            </FireteamTarget>
                            <Connect(TagAreaNameComponent)
                              tag={null}
                            >
                              <TagAreaNameComponent
                                dispatch={[Function]}
                                tag={null}
                              >
                                <span />
                              </TagAreaNameComponent>
                            </Connect(TagAreaNameComponent)>
                          </div>
                        </div>
                      </div>
                    </ActiveFireteam>
                  </Connect(ActiveFireteam)>
                </div>
              </div>
            </Panel>
          </div>
          <div
            key="FT_2"
            style={
              Object {
                "flex": 1,
                "margin": 5,
              }
            }
          >
            <Panel
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "cursor": "pointer",
                  "minWidth": 250,
                }
              }
            >
              <div
                className="panel panel-default"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "cursor": "pointer",
                    "minWidth": 250,
                  }
                }
              >
                <div
                  className="panel-body"
                  style={
                    Object {
                      "padding": 10,
                    }
                  }
                >
                  <Connect(ActiveFireteam)
                    fireteam={
                      Object {
                        "id": "FT_2",
                        "name": "fireteam 2",
                      }
                    }
                  >
                    <ActiveFireteam
                      dispatch={[Function]}
                      fireteam={
                        Object {
                          "id": "FT_2",
                          "name": "fireteam 2",
                        }
                      }
                      fireteamTag={null}
                      targetTag={null}
                    >
                      <div
                        style={
                          Object {
                            "alignItems": "center",
                            "display": "flex",
                            "minHeight": 84,
                          }
                        }
                      >
                        <div
                          style={
                            Object {
                              "flex": 1,
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "display": "flex",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "flex": 1,
                                }
                              }
                            >
                              <span
                                style={
                                  Object {
                                    "fontSize": 20,
                                    "fontWeight": "bold",
                                    "marginRight": 10,
                                  }
                                }
                              >
                                fireteam 2
                              </span>
                              <Connect(TagAreaNameComponent)
                                style={
                                  Object {
                                    "fontSize": 16,
                                  }
                                }
                                tag={null}
                              >
                                <TagAreaNameComponent
                                  dispatch={[Function]}
                                  style={
                                    Object {
                                      "fontSize": 16,
                                    }
                                  }
                                  tag={null}
                                >
                                  <span />
                                </TagAreaNameComponent>
                              </Connect(TagAreaNameComponent)>
                            </div>
                            <div
                              style={
                                Object {
                                  "fontSize": 20,
                                }
                              }
                            >
                              <DistanceBetweenTags
                                from={null}
                                to={null}
                              >
                                <span />
                              </DistanceBetweenTags>
                            </div>
                          </div>
                          <div
                            style={
                              Object {
                                "marginTop": 10,
                              }
                            }
                          >
                            <FireteamTarget
                              fireteamTag={null}
                              targetTag={null}
                            >
                              <span>
                                Nie ustawiono celu
                              </span>
                            </FireteamTarget>
                            <Connect(TagAreaNameComponent)
                              tag={null}
                            >
                              <TagAreaNameComponent
                                dispatch={[Function]}
                                tag={null}
                              >
                                <span />
                              </TagAreaNameComponent>
                            </Connect(TagAreaNameComponent)>
                          </div>
                        </div>
                      </div>
                    </ActiveFireteam>
                  </Connect(ActiveFireteam)>
                </div>
              </div>
            </Panel>
          </div>
        </div>
      </OptionalScroll>
    </FireteamChooser>
  </Connect(FireteamChooser)>
</Provider>
`;
